src/main/java/org/checkerframework/specimin/SpeciminRunner.java

```diff
@@ -155,54 +172,111 @@ private static void performMinimizationImpl(
+   >>> It ensures that the root directory ends with a trailing slash for string manipulation purposes.
+ */
     if (!root.endsWith("/")) {
       root = root + "/";
     }
-
+    
+    /*
+     * =============================================================================STEP 2: Setting Static Solver:====================================================================================
+       >>> This method sets up something called a "symbol solver." 
+           Think of it like a helper that understands and can find definitions of classes, methods, and variables in your code.
+      	   This method sets up a system that helps the parser understand and find definitions in our Java code, 
+           including our project's source code and any additional libraries you are using ( JAVA 17 version libraries).       
+     */
     updateStaticSolver(root, jarPaths);
 
-    // Keys are paths to files, values are parsed ASTs
+ 
+    /*
+     * =============================================================================STEP 3: Parsing Target Files:====================================================================================
+       >>> It parses the target Java files specified in the targetFiles list and stores them in a map (parsedTargetFiles). 
+       	   This map associates file paths <KEY> with parsed Abstract Syntax Trees (ASTs) of the Java files <VALUE>.
+           Abstract syntax trees are data structures widely used in compilers to represent the structure of program code.     
+       >>>  WHAT IS A COMPILATION UNIT?
+              A CompilationUnit is a data structure used in Java programming to represent the entire content of a Java source file.
+              When we use a parser like StaticJavaParser to read a .java file, it creates a CompilationUnit 
+              that represents all the content of that file in a structured way. 
+              This allows our program to understand and work with the code without just treating it as plain text.
+              We can think of it as a digital blueprint of the Java file.
+     */
     Map<String, CompilationUnit> parsedTargetFiles = new HashMap<>();
     for (String targetFile : targetFiles) {
       parsedTargetFiles.put(targetFile, parseJavaFile(root, targetFile));
     }
 
+    /*
+     * ==============================================================================STEP 4: Decompiling JAR Files (Optional):========================================================================
+       >>> If JAR paths are provided (jarPaths list is not empty), it decompiles the JAR files to Java source files in the root directory using the ConsoleDecompiler tool.
+     */
     if (!jarPaths.isEmpty()) {
       List<String> argsToDecompile = new ArrayList<>();
-      argsToDecompile.add("--silent");
-      argsToDecompile.addAll(jarPaths);
-      argsToDecompile.add(root);
+      argsToDecompile.add("--silent");// why is this required?
```

Comment:
I don't think it is required, but the name suggests it turns off some kind of logging from the library. That's generally a good choice: we want to produce our own logs.